src/test/java/org/apache/drillui/test/framework/steps/webui/QuerySteps.java

```diff
@@ -22,31 +22,44 @@
 import org.apache.drillui.test.framework.pages.QueryPage.QueryType;
 
 public final class QuerySteps extends BaseSteps {
+  public enum RESULT_MODE {
+    PHYSICAL,
+    EXEC
+  }
+
+  public QueryResultsSteps runLimitedSQL(String queryText, String resultRowsCount) {
+    BasePage.getPage(QueryPage.class)
+        .setQueryType(QueryType.SQL)
+        .limitQueryResults(resultRowsCount)
+        .submitQuery(queryText);
+    return BaseSteps.getSteps(QueryResultsSteps.class);
+  }
 
   public QueryResultsSteps runSQL(String queryText) {
-    BasePage.getPage(NavigationPage.class)
-        .navigateQuery()
+    BasePage.getPage(QueryPage.class)
         .setQueryType(QueryType.SQL)
         .submitQuery(queryText);
     return BaseSteps.getSteps(QueryResultsSteps.class);
   }
 
   public QueryResultsSteps runPhysical(String queryText) {
-    BasePage.getPage(NavigationPage.class)
-        .navigateQuery()
+    BasePage.getPage(QueryPage.class)
         .setQueryType(QueryType.PHYSICAL)
         .submitQuery(queryText);
     return BaseSteps.getSteps(QueryResultsSteps.class);
   }
 
   public QueryResultsSteps runLogical(String queryText) {
-    BasePage.getPage(NavigationPage.class)
-        .navigateQuery()
+    BasePage.getPage(QueryPage.class)
         .setQueryType(QueryType.LOGICAL)
         .submitQuery(queryText);
     return BaseSteps.getSteps(QueryResultsSteps.class);
   }
 
+  public static String prepareLogicalPlan(String queryText, RESULT_MODE result_mode) {
+    return queryText.replace("\"resultMode\" : \"LOGICAL\"", "\"resultMode\" : \"" + result_mode + "\"");
```

Comment:
Will be enough:
```suggestion
    return queryText.replace("\"LOGICAL\"", "\"" + result_mode + "\"");
```

Reply:
Some other part of plan can contains "LOGICAL" part. It may cause a problems.